src/capi.rs

```diff
@@ -464,23 +489,42 @@ pub unsafe extern fn rav1e_config_parse_int(
   }
 }
 
+fn context_new_internal(
+  cfg: &rav1e::Config,
+) -> Result<Context, rav1e::EncoderStatus> {
+  let enc = &cfg.enc;
+  Ok(Context {
+    ctx: match enc.bit_depth {
+      8 => EncContext::U8(cfg.new_context()?),
+      _ => EncContext::U16(cfg.new_context()?),
+    },
+    last_err: None,
+  })
+}
+
 /// Generate a new encoding context from a populated encoder configuration
 ///
 /// Multiple contexts can be generated through it.
+/// Returns Null if context creation failed, e.g. by passing
+/// an invalid Config.
 #[no_mangle]
 pub unsafe extern fn rav1e_context_new(cfg: *const Config) -> *mut Context {
   let cfg = &(*cfg).cfg;
   let enc = &cfg.enc;
 
-  let ctx = Context {
+  let ctx_maybe = Result::<Context> {
```

Comment:
This should call `context_new_internal` I guess.

Reply:
`context_new_internal` Seems to be added and never used?

Reply:
Yeah I meant to drop context_new_internal entirely.